2021/01/27 14:06:38 : Traceback (most recent call last):
  File "/home/phuonglc/project/project_v32/cartmigration/controllers/migration.py", line 665, in storage_data
    getattr(self.get_source_cart(), 'finish_storage_data')()
  File "/home/phuonglc/project/project_v32/cartmigration/models/basecart.py", line 3932, in finish_storage_data
    folder_upload = self.get_folder_upload(self._migration_id)
  File "/home/phuonglc/project/project_v32/cartmigration/models/basecart.py", line 3944, in get_folder_upload
    folder_upload = root_path + '/' + DIR_UPLOAD + '/' + get_config_ini('src', 'file', file = 'test.ini')
TypeError: can only concatenate str (not "NoneType") to str

2021/01/27 14:06:41 : categories_errors 2: Traceback (most recent call last):
  File "/home/phuonglc/project/project_v32/cartmigration/controllers/migration.py", line 1072, in migration
    import_data = fn_import(convert_data, main, ext)
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/woocommerce.py", line 2278, in category_import
    "count":len(convert['categories_ext']['products_to_categories'])
KeyError: 'products_to_categories'

2021/01/27 14:06:41 : categories_errors 3: Traceback (most recent call last):
  File "/home/phuonglc/project/project_v32/cartmigration/controllers/migration.py", line 1072, in migration
    import_data = fn_import(convert_data, main, ext)
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/woocommerce.py", line 2278, in category_import
    "count":len(convert['categories_ext']['products_to_categories'])
KeyError: 'products_to_categories'

2021/01/27 14:06:41 : categories_errors 4: Traceback (most recent call last):
  File "/home/phuonglc/project/project_v32/cartmigration/controllers/migration.py", line 1072, in migration
    import_data = fn_import(convert_data, main, ext)
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/woocommerce.py", line 2278, in category_import
    "count":len(convert['categories_ext']['products_to_categories'])
KeyError: 'products_to_categories'

2021/01/27 14:06:41 : categories_errors 5: Traceback (most recent call last):
  File "/home/phuonglc/project/project_v32/cartmigration/controllers/migration.py", line 1072, in migration
    import_data = fn_import(convert_data, main, ext)
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/woocommerce.py", line 2278, in category_import
    "count":len(convert['categories_ext']['products_to_categories'])
KeyError: 'products_to_categories'

2021/01/27 14:06:41 : categories_errors 6: Traceback (most recent call last):
  File "/home/phuonglc/project/project_v32/cartmigration/controllers/migration.py", line 1072, in migration
    import_data = fn_import(convert_data, main, ext)
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/woocommerce.py", line 2278, in category_import
    "count":len(convert['categories_ext']['products_to_categories'])
KeyError: 'products_to_categories'

2021/01/27 14:06:41 : categories_errors 7: Traceback (most recent call last):
  File "/home/phuonglc/project/project_v32/cartmigration/controllers/migration.py", line 1072, in migration
    import_data = fn_import(convert_data, main, ext)
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/woocommerce.py", line 2278, in category_import
    "count":len(convert['categories_ext']['products_to_categories'])
KeyError: 'products_to_categories'

2021/01/27 14:06:41 : categories_errors 8: Traceback (most recent call last):
  File "/home/phuonglc/project/project_v32/cartmigration/controllers/migration.py", line 1072, in migration
    import_data = fn_import(convert_data, main, ext)
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/woocommerce.py", line 2286, in category_import
    parent_id = self.import_category_parent(convert['parent'])
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/woocommerce.py", line 5107, in import_category_parent
    category_parent_import = self.category_import(convert_parent, category, categories_ext)
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/woocommerce.py", line 2278, in category_import
    "count":len(convert['categories_ext']['products_to_categories'])
KeyError: 'products_to_categories'

2021/01/27 14:06:41 : categories_errors 9: Traceback (most recent call last):
  File "/home/phuonglc/project/project_v32/cartmigration/controllers/migration.py", line 1072, in migration
    import_data = fn_import(convert_data, main, ext)
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/woocommerce.py", line 2278, in category_import
    "count":len(convert['categories_ext']['products_to_categories'])
KeyError: 'products_to_categories'

2021/01/27 14:06:41 : categories_errors 10: Traceback (most recent call last):
  File "/home/phuonglc/project/project_v32/cartmigration/controllers/migration.py", line 1072, in migration
    import_data = fn_import(convert_data, main, ext)
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/woocommerce.py", line 2278, in category_import
    "count":len(convert['categories_ext']['products_to_categories'])
KeyError: 'products_to_categories'

2021/01/27 14:06:41 : categories_errors 11: Traceback (most recent call last):
  File "/home/phuonglc/project/project_v32/cartmigration/controllers/migration.py", line 1072, in migration
    import_data = fn_import(convert_data, main, ext)
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/woocommerce.py", line 2278, in category_import
    "count":len(convert['categories_ext']['products_to_categories'])
KeyError: 'products_to_categories'

2021/01/27 14:06:41 : categories_errors 12: Traceback (most recent call last):
  File "/home/phuonglc/project/project_v32/cartmigration/controllers/migration.py", line 1072, in migration
    import_data = fn_import(convert_data, main, ext)
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/woocommerce.py", line 2278, in category_import
    "count":len(convert['categories_ext']['products_to_categories'])
KeyError: 'products_to_categories'

2021/01/27 14:06:41 : categories_errors 13: Traceback (most recent call last):
  File "/home/phuonglc/project/project_v32/cartmigration/controllers/migration.py", line 1072, in migration
    import_data = fn_import(convert_data, main, ext)
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/woocommerce.py", line 2286, in category_import
    parent_id = self.import_category_parent(convert['parent'])
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/woocommerce.py", line 5107, in import_category_parent
    category_parent_import = self.category_import(convert_parent, category, categories_ext)
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/woocommerce.py", line 2278, in category_import
    "count":len(convert['categories_ext']['products_to_categories'])
KeyError: 'products_to_categories'

2021/01/27 14:06:41 : categories_errors 14: Traceback (most recent call last):
  File "/home/phuonglc/project/project_v32/cartmigration/controllers/migration.py", line 1072, in migration
    import_data = fn_import(convert_data, main, ext)
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/woocommerce.py", line 2286, in category_import
    parent_id = self.import_category_parent(convert['parent'])
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/woocommerce.py", line 5107, in import_category_parent
    category_parent_import = self.category_import(convert_parent, category, categories_ext)
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/woocommerce.py", line 2278, in category_import
    "count":len(convert['categories_ext']['products_to_categories'])
KeyError: 'products_to_categories'

2021/01/27 14:06:41 : categories_errors 15: Traceback (most recent call last):
  File "/home/phuonglc/project/project_v32/cartmigration/controllers/migration.py", line 1072, in migration
    import_data = fn_import(convert_data, main, ext)
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/woocommerce.py", line 2278, in category_import
    "count":len(convert['categories_ext']['products_to_categories'])
KeyError: 'products_to_categories'

2021/01/27 14:06:41 : categories_errors 16: Traceback (most recent call last):
  File "/home/phuonglc/project/project_v32/cartmigration/controllers/migration.py", line 1072, in migration
    import_data = fn_import(convert_data, main, ext)
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/woocommerce.py", line 2278, in category_import
    "count":len(convert['categories_ext']['products_to_categories'])
KeyError: 'products_to_categories'

2021/01/27 14:06:42 : categories_errors 17: Traceback (most recent call last):
  File "/home/phuonglc/project/project_v32/cartmigration/controllers/migration.py", line 1072, in migration
    import_data = fn_import(convert_data, main, ext)
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/woocommerce.py", line 2286, in category_import
    parent_id = self.import_category_parent(convert['parent'])
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/woocommerce.py", line 5107, in import_category_parent
    category_parent_import = self.category_import(convert_parent, category, categories_ext)
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/woocommerce.py", line 2278, in category_import
    "count":len(convert['categories_ext']['products_to_categories'])
KeyError: 'products_to_categories'

2021/01/27 14:06:42 : categories_errors 19: Traceback (most recent call last):
  File "/home/phuonglc/project/project_v32/cartmigration/controllers/migration.py", line 1072, in migration
    import_data = fn_import(convert_data, main, ext)
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/woocommerce.py", line 2278, in category_import
    "count":len(convert['categories_ext']['products_to_categories'])
KeyError: 'products_to_categories'

2021/01/27 14:06:42 : categories_errors 20: Traceback (most recent call last):
  File "/home/phuonglc/project/project_v32/cartmigration/controllers/migration.py", line 1072, in migration
    import_data = fn_import(convert_data, main, ext)
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/woocommerce.py", line 2278, in category_import
    "count":len(convert['categories_ext']['products_to_categories'])
KeyError: 'products_to_categories'

2021/01/27 14:06:42 : categories_errors 22: Traceback (most recent call last):
  File "/home/phuonglc/project/project_v32/cartmigration/controllers/migration.py", line 1072, in migration
    import_data = fn_import(convert_data, main, ext)
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/woocommerce.py", line 2278, in category_import
    "count":len(convert['categories_ext']['products_to_categories'])
KeyError: 'products_to_categories'

2021/01/27 14:06:42 : categories_errors 23: Traceback (most recent call last):
  File "/home/phuonglc/project/project_v32/cartmigration/controllers/migration.py", line 1072, in migration
    import_data = fn_import(convert_data, main, ext)
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/woocommerce.py", line 2278, in category_import
    "count":len(convert['categories_ext']['products_to_categories'])
KeyError: 'products_to_categories'

2021/01/27 14:06:53 : orders_errors 8: Traceback (most recent call last):
  File "/home/phuonglc/project/project_v32/cartmigration/controllers/migration.py", line 1036, in migration
    convert = fn_convert_export(main, ext)
  File "/home/phuonglc/project/project_v32/cartmigration/models/cart/custom.py", line 1160, in convert_order_export
    shipping_details = [x for x in data if x['class'] == 'ot_shipping'][0]
IndexError: list index out of range

